### PR TITLE
Add unit tests to indexer.js in Parser

### LINKED ISSUE #3224: Tests for the ElasticSearch Indexer in the backend and Parser
In #1417 , the ES indexer will be modified or a new indexer will be added to include analyzers in order to setup for autocomplete on Authors. 

Currently in the legacy backend, the [indexer.js](https://github.com/Seneca-CDOT/telescope/blob/master/src/backend/utils/indexer.js) module is mocked as a whole in `__mocks__`. 
In the new Parser tests (#3040) this manual mock will be taken out, and other tests that exports from the Parser [indexer.js](https://github.com/Seneca-CDOT/telescope/blob/master/src/api/parser/src/utils/indexer.js) will make use of ElasticSearch-mock. 
In either case, we should add a test file for the `indexer.js` module to make sure our indexing works as expected.

### PR DESCRIPTION
**Pull request by rclee91**
## Issue This PR Addresses
Fixes #3224 

## Description

Add unit tests for [src/api/parser/src/utils/indexer.js](https://github.com/Seneca-CDOT/telescope/blob/master/src/api/parser/src/utils/indexer.js)

## Steps to test the PR

`pnpm jest indexer`

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not

### CODE REVIEW

File: src/api/parser/test/indexer.test.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
+const { POSTS_URL = 'http://localhost/v1/posts' } = process.env;
+const { Elastic, logger, createError } = require('@senecacdot/satellite');
+const { waitOnReady, indexPost, deletePost, search } = require('../src/utils/indexer');
+
+const index = 'posts';
+const { mock } = Elastic();
+const loggerSpy = jest.spyOn(logger, 'error').mockImplementation((msg) => msg);
+
+afterEach(() => {
```
Comment (humphd) [inline src/api/parser/test/indexer.test.js:9]: Can you move this inside the `describe()` below?

File: src/api/parser/test/indexer.test.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  const createSpy = jest.fn().mockReturnValue({});
+  it('Established connection and successful index creation should have no errors or logged errors', async () => {
+    mock.add(
+      {
+        method: 'PUT',
+        path: `/${index}`,
+      },
+      createSpy
+    );
+    await waitOnReady();
```
Comment (humphd) [inline src/api/parser/test/indexer.test.js:40]: Do this once in a `beforeEach()`

File: src/api/parser/test/indexer.test.js (hunk 1 above)
Comment (humphd) [reply src/api/parser/test/indexer.test.js:9]: Readying your tests, you can wrap the entire thing in a `describe()` and put it there.

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/api/parser/test/indexer.test.js (hunk 2 above)
Comment (rclee91) [reply src/api/parser/test/indexer.test.js:40]: I'm not sure which part you're referring to, is it for `await waitOnReady()` ? 
The two mocks in this describe are slightly different. I need to set up the mocks before calling `waitOnReady()`, cause `createPostsIndex()` is called in there.

Comment (humphd) [reply src/api/parser/test/indexer.test.js:40]: Ah, it seemed you were doing the same pattern multiple times, which is a "bad code smell" and often means you can refactor.  If you need to repeat, do it for sure.

Comment (rclee91) [reply src/api/parser/test/indexer.test.js]: I've changed it so `mock.add()` is wrapped in a function that can be called for each unit test.

**Review by humphd — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]

**Pull request merged**